scylla-cdc/src/consumer.rs

```diff
@@ -0,0 +1,194 @@
+const TTL_NAME: &str = "cdc$ttl";
+const IS_DELETED_PREFIX: &str = "cdc$deleted_";
+const ARE_ELEMENTS_DELETED_PREFIX: &str = "cdc$deleted_elements_";
+
+impl CDCRowSchema {
+    pub fn new(specs: &[ColumnSpec]) -> CDCRowSchema {
+        // [TODO]:  This function is not usable now.
+        //          The driver does not provide the vector of ColumnSpec for every query yet,
+        //          and therefore we can't map the result
+        //          to something other than a hardcoded tuple.
+
+        let mut mapping: HashMap<String, usize> = HashMap::new();
+        let mut deleted_mapping: HashMap<String, usize> = HashMap::new();
+        let mut deleted_el_mapping: HashMap<String, usize> = HashMap::new();
+        let (mut stream_id, mut time, mut batch_seq_no, mut end_of_batch, mut operation, mut ttl):
+            (usize, usize, usize, usize, usize, usize) = (0, 0, 0, 0, 0, 0);
+        let mut j = 0;
+
+        // Hashmaps will have indices of data in a new vector without the hardcoded values.
+        for (i, spec) in specs.iter().enumerate() {
+            match spec.name.as_str() {
+                // spec.name is public since 0.3.0 driver version, it didn't work on 0.2.1.
+                STREAM_ID_NAME => stream_id = i,
+                TIME_NAME => time = i,
+                BATCH_SEQ_NO_NAME => batch_seq_no = i,
+                END_OF_BATCH_NAME => end_of_batch = i,
+                OPERATION_NAME => operation = i,
+                TTL_NAME => ttl = i,
+                x => {
+                    if x.starts_with(IS_DELETED_PREFIX) {
+                        deleted_mapping.insert(x.to_string(), j);
+                    } else if x.starts_with(ARE_ELEMENTS_DELETED_PREFIX) {
+                        deleted_el_mapping.insert(x.to_string(), j);
+                    } else {
+                        mapping.insert(x.to_string(), j);
+                    }
+                    j += 1;
+                }
+            }
+        }
+
+        CDCRowSchema {
+            stream_id,
+            time,
+            batch_seq_no,
+            end_of_batch,
+            operation,
+            ttl,
+            mapping,
+            deleted_mapping,
+            deleted_el_mapping,
+        }
+    }
+}
+
+pub struct CDCRow {
+    pub stream_id: Vec<u8>,
+    pub time: uuid::Uuid,
+    pub batch_seq_no: i32,
+    pub end_of_batch: bool,
+    pub operation: OperationType,
+    // Can be NULL in the database.
+    pub ttl: Option<i64>,
+    data: Vec<Option<CqlValue>>,
+    // Maps element name to its index in the data vector.
+    mapping: HashMap<String, usize>,
+    deleted_mapping: HashMap<String, usize>,
+    deleted_el_mapping: HashMap<String, usize>,
+}
+
+impl CDCRow {
+    pub fn from_row(row: Row, schema: &CDCRowSchema) -> CDCRow {
+        let mapping = schema.mapping.clone();
+        let deleted_mapping = schema.deleted_mapping.clone();
+        let deleted_el_mapping = schema.deleted_el_mapping.clone();
```

Comment:
It's really sad that construction of a CDCRow requires cloning three hashmaps.

Let's change the CDCRow so that it keeps a reference to a CDCRowSchema:

```rust
pub struct CDCRow<'schema>
    // ...
    schema: &'schema CDCRowSchema,
```

Then, in the getter methods you can refer to the mappings through the `schema` reference.